dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/MetricAdapterTest.java

```diff
@@ -168,4 +168,113 @@ public void TestAddOneAgentDimensionsEmptyList() {
                     list))
             .serialize());
   }
+
+  @Test
+  public void TestOneAgentDimensionsOverwrite() {
+    Collection<AbstractMap.SimpleEntry<String, String>> tags =
+        Arrays.asList(
+            new AbstractMap.SimpleEntry<>("oneagenttag", "oneagentvalue"),
+            new AbstractMap.SimpleEntry<>("oneagenttag2", "oneagentvalue2"));
+    MetricAdapter.getInstance().setTags(tags);
+
+    List<ValueAtPercentile> list = new ArrayList<>(2);
+    list.add(ValueAtPercentile.create(0.0, 1.56));
+    list.add(ValueAtPercentile.create(100.0, 345.23));
+    assertEquals(
+        "metric_01,oneagenttag=oneagentvalue,oneagenttag2=oneagentvalue2 gauge,min=1.56,max=345.23,sum=12934.0,count=42 456",
+        MetricAdapter.generateSummaryPoint(
+                "metric_01",
+                DoubleSummaryPointData.create(
+                    123,
+                    TimeUnit.MILLISECONDS.toNanos(456),
+                    // these will be overritten by the tags set in the singleton.
+                    Labels.of("oneAgentTag", "someValue", "oneAgentTag2", "someOtherValue"),
+                    42,
+                    12934,
+                    list))
+            .serialize());
+  }
+
+  @Test
+  public void Test_getUniqueCombinedDimensions_Valid() {
+    Labels labels = Labels.of("dim1", "dv1", "dim2", "dv2");
+    Collection<AbstractMap.SimpleEntry<String, String>> tags =
+        Arrays.asList(
+            new AbstractMap.SimpleEntry<>("tag1", "tv1"),
+            new AbstractMap.SimpleEntry<>("tag2", "tv2"));
+    MetricAdapter.getInstance().setTags(tags);
+
+    List<Dimension> expected =
+        Arrays.asList(
+            Dimension.create("dim1", "dv1"),
+            Dimension.create("dim2", "dv2"),
+            Dimension.create("tag1", "tv1"),
+            Dimension.create("tag2", "tv2"));
+    // the array list wrap is so the list is sortable as it is otherwise unmodifiable.
+    List<Dimension> got = new ArrayList<>(MetricAdapter.getUniqueCombinedDimensions(labels));
```

Comment:
```suggestion
    List<Dimension> actual = new ArrayList<>(MetricAdapter.getUniqueCombinedDimensions(labels));
```
is commonly used for that